src/UsgsAstroFrameSensorModel.cpp

```diff
@@ -821,71 +824,68 @@ std::string UsgsAstroFrameSensorModel::constructStateFromIsd(const std::string&
-        state["m_currentParameterValue"][1] = metric_conversion(state["m_currentParameterValue"][1].get<double>(), unit);
-        state["m_currentParameterValue"][2] = metric_conversion(state["m_currentParameterValue"][2].get<double>(), unit);
-      }
+        state["m_currentParameterValue"] = json();
+        state["m_currentParameterValue"][0] = metric_conversion(positions[0].get<double>(), unit);
+        state["m_currentParameterValue"][1] = metric_conversion(positions[1].get<double>(), unit);
+        state["m_currentParameterValue"][2] = metric_conversion(positions[2].get<double>(), unit);
+        state["m_spacecraftVelocity"] = velocities;
 
-      // get  sensor_velocity
-      {
-        json jayson = isd.at("sensor_velocity");
-        json x = jayson.at("x");
-        json y = jayson.at("y");
-        json z = jayson.at("z");
-
-        state["m_spacecraftVelocity"] = json();
-        state["m_spacecraftVelocity"][0] = x;
-        state["m_spacecraftVelocity"][1] = y;
-        state["m_spacecraftVelocity"][2] = z;
+        std::cerr << "Sensor Location Parsed!" << std::endl;
       }
 
       // get sun_position
+      // sun position is not strictly necessary, but is required for getIlluminationDirection.
       {
         json jayson = isd.at("sun_position");
-        json x = jayson.at("x");
-        json y = jayson.at("y");
-        json z = jayson.at("z");
+        json positions = jayson.at("positions")[0];
+        json unit = jayson.at("unit");
 
-        state["m_sunPosition"][0] = x;
-        state["m_sunPosition"][1] = y;
-        state["m_sunPosition"][2] = z;
+        unit = unit.get<std::string>();
+        state["m_sunPosition"] = json();
+        state["m_sunPosition"][0] = metric_conversion(positions[0].get<double>(), unit);
+        state["m_sunPosition"][1] = metric_conversion(positions[1].get<double>(), unit);
+        state["m_sunPosition"][2] = metric_conversion(positions[2].get<double>(), unit);
+
+        std::cerr << "Sun Position Parsed!" << std::endl;
       }
 
-      // get sun_position
-      // sun position is not strictly necessary, but is required for getIlluminationDirection.
+      // get sensor_orientation quaternion
       {
-        state["m_currentParameterValue"][3] = isd.at("sensor_orientation").at(0);
-        state["m_currentParameterValue"][4] = isd.at("sensor_orientation").at(1);
-        state["m_currentParameterValue"][5] = isd.at("sensor_orientation").at(2);
-        state["m_currentParameterValue"][6] = isd.at("sensor_orientation").at(3);
+        json jayson = isd.at("sensor_orientation");
+        json quaternion = jayson.at("quaternions")[0];
+
+        state["m_currentParameterValue"][3] = quaternion[0];
+        state["m_currentParameterValue"][4] = quaternion[1];
+        state["m_currentParameterValue"][5] = quaternion[2];
+        state["m_currentParameterValue"][6] = quaternion[3];
```

Comment:
So this looks like an array of arrays in form `[[<all 0>], [<all 1>], [<all 2>], [<all 3>]]`. The specification is `[[q1], [q2], [q3], ..., [qn]]`. Is this setting the parameter value for just the first quaternion.

Reply:
Yes, line 867, pull off the first set of quaternions (q1) and then this parses those into the current parameters

Reply:
:+1: